pkg/sip/inbound.go

```diff
@@ -827,7 +851,7 @@ func (c *inboundCall) close(error bool, status CallStatus, reason string) {
 
 	// Call the handler asynchronously to avoid blocking
 	if c.s.handler != nil {
-		go c.s.handler.OnCallEnd(context.Background(), c.state.callInfo, reason)
+		c.s.handleSessionEnd(context.Background(), c.call, c.state, c.state.callInfo.ParticipantAttributes[AttrSIPCallIDFull], reason, c.projectID)
```

Comment:
Same note for the call ID as below: it should be a field on a call, getting things from attributes should be avoided.